metecho/api/views.py

```diff
@@ -198,6 +213,62 @@ class UserViewSet(mixins.RetrieveModelMixin, mixins.ListModelMixin, GenericViewS
     queryset = User.objects.all()
 
 
+class ProjectDependencyViewSet(ReadOnlyModelViewSet):
+    """Dependencies available during Project creation"""
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = ProjectDependencySerializer
+    pagination_class = CustomPaginator
+    queryset = ProjectDependency.objects.all()
+
+
+class GitHubOrganizationViewSet(ReadOnlyModelViewSet):
+    """GitHub Organizations available during Project creation"""
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = GitHubOrganizationSerializer
+    pagination_class = CustomPaginator
+    queryset = GitHubOrganization.objects.all()
+
+    @extend_schema(request=None, responses=GitHubUserMinimalSerializer(many=True))
+    @action(detail=True, methods=["GET"])
+    def members(self, request, pk):
+        """Fetch the members of an Organization from GitHub"""
+        org: GitHubOrganization = self.get_object()
+        gh_api = gh.gh_given_user(request.user)
+        gh_org = gh_api.organization(org.login)
+        members = sorted(
+            (member.as_dict() for member in gh_org.members()),
+            key=lambda member: member["login"].lower(),
+        )
+        members = GitHubUserMinimalSerializer(data=members, many=True)
+        members.is_valid(raise_exception=True)
+        return Response(members.data)
+
+    @extend_schema(
+        request=CheckRepoNameSerializer,
+        responses=OpenApiResponse(
+            {"type": "object", "properties": {"available": {"type": "boolean"}}},
+            description="",
+        ),
+    )
+    @action(detail=True, methods=["POST"])
+    def check_repo_name(self, request, pk):
+        """Determine if a repository name is available for the Organization on GitHub"""
+        org: GitHubOrganization = self.get_object()
+        serializer = CheckRepoNameSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        try:
+            gh.get_repo_info(
+                None, repo_owner=org.login, repo_name=serializer.validated_data["name"]
+            )
+        except NotFoundError:
+            available = True
+        else:
+            available = False
```

Comment:
So this only checks repos that the GH app has access to. For example, with OddBird "books" will return false, but "oddcontrast" will return true -- the local dev GH app has access to Books, but not to OddContrast. 🤔 

I'm not sure if there's a way around that -- or what will happen if I try to create a repo that already exists? Maybe there's a better way we can check repo names?

Reply:
Hmm, that's a tough one. I'm not sure we can effectively check the name in this case. I'll see what happens if we actually try to create a repo with a duplicate name

Reply:
Yeah, I suppose at the very least we need to be sure we're handling the error on create correctly. I guess if our GH app doesn't have access, there's no good way to know about private repos?

Reply:
> there's no good way to know about private repos?

Yep, I agree.

Regarding error handling, it looks like the API calls fail with 422 errors. This will cause the Project to be deleted from Metecho (it does the same for all exceptions):

- When the team name is taken: `github3.exceptions.UnprocessableEntity: 422 Validation Failed`
- When the repo slug is taken: `github3.exceptions.UnprocessableEntity: 422 Repository creation failed`

Reply:
Ok, that seems reasonable. I think we should confirm with SFDO whether we want additional help-text or changed functionality or something -- it seems potentially confusing for us to say "this name is available" if we actually can't say whether a given name is available or not.

Reply:
SFDO confirmed we want to use a single GH app, and we should check it has access to all repos before proceeding